api/types.proto

```diff
@@ -30,12 +30,31 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
+message Set {
+	message Void {}
+	map<string, Void> val = 1;
+}
+
+message Integer {
+	uint64 val = 1;
+}
+
+message ThirdPartyResource {
```

Comment:
I discussed naming with some of the other maintainers, and we liked either of the names `AbstractResource` or `GenericResource`.

We already use the term "generic" in some places, for example `GenericRuntimeSpec`. I'm not entirely sure whether this is an argument for or against using `Generic` here. `GenericRuntimeSpec` is a message that contains an `Any` message, so what's "generic" about it is that the user can embed their own data. The resource concept is a bit different.

Do either of those names sound good to you? Do you have a preference?